### ISSUE TITLE
Ng Serve Fails To Compile After Modifying Interfaces/Classes And Accessing New Properties

### ISSUE THREAD

**Issue by jamesqquick** [angular/angular-cli#8807]
### Versions
4.4.6

### Repro steps

* Step 1 - Start app with Ng Serve 
* Step 2 - Create interface with some properties and save
* Step 3 - Update the interface to add a new property, ex. 'turtle'
* Step 4 - Update a component or service to access this new property

### Observed behavior
Visual Studio Code does not display and error but when ng serve rebuilds, it throws an error that the new property does not exist.
```
ERROR in C:/Users/884170/Desktop/ILS_UI_Angular4-James/src/app/components/tracking-number-lookup/tracking-number-lookup.component.ts (142,15): Property 'turtle' does not exist on type 'EvaluatedShipmentStatus'.
ERROR in C:/Users/884170/Desktop/ILS_UI_Angular4-James/src/app/components/tracking-number-lookup/tracking-number-lookup.component.ts (142,15): Property 'turtle' does not exist on type 'EvaluatedShipmentStatus'.
```

### Desired behavior
I would like to see the Angular CLI build system be able to recognize updated to an interface/class and build successfully when added properties are accessed.  

I frequently am updating models and then accessing the new properties.  Each time I do this, I have to completely stop ng serve and restart it.

**Comment by Brocco** [comment]
Can you please include the version of the Angular CLI that you're using in your project?

**Comment by jamesqquick** [comment]
When I did an 'ng --version' I got 4.4.6 for everything including angular cli

**Comment by filipesilva** [comment]
I tried to reproduce your issue with the latest version of Angular CLI and Angular, but could. This is what I did:

- start `ng serve`
- add interface to `src/app/app.module.ts` and save:
```
export interface ITurtle {
  notTurtle: number;
  turtle: number;
}
```
- import, use interface in `src/app/app.component.ts` and save file:
```
import { ITurtle } from "./app.module";

const turtle: ITurtle = {
  notTurtle: 1,
  turtle: 2
}
```
- all rebuilds were successful without errors.

I think this happens on your case because you're still on Angular 4. We have an updated TS compilation pipeline starting with Angular 5.0 and CLI 1.5, that should have addressed https://github.com/angular/angular-cli/issues/2034.

If you have a reproduction with the latest versions let me know and I will reopen.
